api/gateway-operator/v1beta1/gatewayconfiguration_types.go

```diff
@@ -145,6 +157,39 @@ type GatewayConfigServiceOptions struct {
 	ServiceOptions `json:",inline"`
 }
 
+// GatewayConfigurationListenerOptions specifies configuration overrides of defaults on certain listener of the Gateway.
+// The name must match the name of a listener in the Gateway
+// and the options are applied to the configuration of the matching listener.
+// For example, if the option for listener "http" specified the nodeport number to 30080,
+// The ingress service will expose the nodeport 30080 for the "http" listener of the Gateway.
+// For listeners without an item in listener options of GatewayConfiguration, default configuration is used for it.
+// +apireference:kgo:include
+type GatewayConfigurationListenerOptions struct {
+	// Name is the name of the Listener.
+	//
+	// +required
+	Name gatewayv1.SectionName `json:"name"`
+
+	// The port on each node on which this service is exposed when type is
+	// NodePort or LoadBalancer. Usually assigned by the system. If a value is
+	// specified, in-range, and not in use it will be used, otherwise the
+	// operation will fail. If not specified, a port will be allocated if this
+	// Service requires one. If this field is specified when creating a
+	// Service which does not need it, creation will fail. This field will be
+	// wiped when updating a Service to no longer need it (e.g. changing type
+	// from NodePort to ClusterIP).
+	//
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
+	//
+	// Can only be specified if type of the dataplane ingress service (specified in `spec.dataplaneOptions.network.services.ingress.type`)
+	// is NodePort or LoadBalancer.
+	//
+	// +optional
+	//
+	// +optional
+	NodePort *int32 `json:"nodePort,omitempty"`
```

Comment:
Since we currently have no other options, I would mark this as required rather than optional. In the future, when multiple options become available, we can relax this requirement.